02-crud.c

```diff
@@ -0,0 +1,208 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+
+#define FILENAME "users.txt"
+#define MAX_NAME_LEN 100
+
+typedef struct
+{
+    char userId[20];
+    char name[MAX_NAME_LEN];
+    int age;
+} User;
+
+void createUser()
+{
+    // a+->allow both reading and appending.
+    FILE *file = fopen(FILENAME, "a+");
+    if (!file)
+    {
+        printf("Unable to open or create the file.\n");
+        return;
+    }
+    User user;
+    printf("Enter unique ID: ");
+    scanf("%s", user.userId);
+    // set file to intial position
+    rewind(file);
+
+    // Here we first check the user exist with that id or not.
+    char userId[20];
+    while (fscanf(file, "%s", userId) != EOF)
+    {
+        if (strcmp(userId, user.userId) == 0)
+        {
+            printf("User ID already exists.\n");
+            return;
+        }
+    }
+    fclose(file);
+    file = fopen(FILENAME, "a");
+
+    printf("Enter FirstName: ");
+    scanf("%s", user.name);
+    printf("Enter age: ");
+    scanf("%d", &user.age);
+
+    fprintf(file, "%s %s %d\n", user.userId, user.name, user.age);
+    fclose(file);
+    printf("User added successfully!\n");
+}
+
+void readUsers()
+{
+    FILE *file = fopen(FILENAME, "r");
+    if (!file)
+    {
+        printf("No users found.\n");
+        return;
+    }
+    User user;
+    printf("Users:\n");
+    // MACRO EOF return when we encounter with a  special character whose ASCII code 26 indicate end of file.
+    int i = 1;
+    while (fscanf(file, "%s %s %d", user.userId, user.name, &user.age) != EOF)
+    {
+        printf("User: %d,ID: %s, Name: %s, Age: %d\n", i++, user.userId, user.name, user.age);
+    }
+
+    fclose(file);
+}
+
+void updateFileStatus(bool found, const char *searchId, const char *msg)
+{
+    if (found)
+    {
+        remove(FILENAME);
+        rename("temp.txt", FILENAME);
+        printf("%s", msg);
+    }
+    else
+    {
+        remove("temp.txt");
+        printf("User with ID %s not found.\n", searchId);
+    }
+}
+bool updateUser(FILE *file, FILE *tempFile, const char *searchId)
```

Comment:
Too much of logic in single function.
Make small functions according to particular logics and use them here.